firefox_media_tests/playback/test_video_playback.py

```diff
@@ -0,0 +1,57 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+from marionette_driver.errors import TimeoutException
+from marionette_driver import Wait
+
+from firefox_media_tests.utils import (verbose_until)
+from media_test_harness.testcase import MediaTestCase
+from media_utils.video_puppeteer import (playback_done, playback_started,
+                                         VideoPuppeteer, VideoException)
+
+
+class TestVideoPlaybackBase(MediaTestCase):
+
+    def setUp(self):
+        MediaTestCase.setUp(self)
+        self.test_urls = self.video_urls
+
+    def tearDown(self):
+        MediaTestCase.tearDown(self)
+
+    def test_playback_starts(self):
+        with self.marionette.using_context('content'):
+            for url in self.test_urls:
+                video = VideoPuppeteer(self.marionette, url)
+                verbose_until(Wait(video, timeout=30), video,
+                              playback_started)
+                video.player_pause()
+
+    def test_video_playback_sanity(self):
+        self.run_playback(timeout=120)
+
+    def test_video_playback_full(self):
+        self.run_playback()
+
+    def run_playback(self, timeout=None):
+        with self.marionette.using_context('content'):
+            for url in self.test_urls:
+                video = VideoPuppeteer(self.marionette, url)
+                verbose_until(Wait(video, timeout=30), video,
+                              playback_started)
+                duration_timeout = video.player_duration * 1.3
+                if timeout is None:
+                    duration = duration_timeout
+                else:
+                    duration = min(timeout, duration_timeout)
+
+                try:
+                    verbose_until(Wait(video, interval=5, timeout=duration),
+                                  video, playback_done)
+                except TimeoutException as e:
+                    pass
```

Comment:
That's how I limit the amount of playback time for the sanity check. The video will either still be playing when the duration is reached, will have stopped (and this iteration of the loop exited), or will have thrown some other exception. We know the video started because we waited for it to start above.